src/main/java/org/fundacionjala/at13/spaceinvaders/GameBoard.java

```diff
@@ -79,27 +131,39 @@ public void keyPressed(final KeyEvent e) {
             spaceship.moveRight();
             refresh();
         }
+        if (e.getKeyCode() == KeyEvent.VK_SPACE) {
+            if (!switchBullet) {
+                actualPositionX = spaceship.getPosX();
+            }
+            switchBullet = true;
+        }
     }
 
     /**
-    * @Override keyReleased.
-    */
+     * @Override keyReleased.
+     */
     @Override
     public void keyReleased(final KeyEvent e) {
         return;
     }
 
     /**
      * Refresh the icon of a label where it is our spaceship.
-     * */
+     */
     public void refresh() {
         ImageIcon iconLogo = new ImageIcon("resources/spaceship.png");
         labelArray[spaceship.getPosY()][spaceship.getPosX()].setIcon(iconLogo);
+
```

Comment:
maybe remove this line break in the code

Reply:
Already solved 😁

Reply:
ok, well done